server/app.ts

```diff
@@ -53,6 +54,41 @@ export default function createApp(services: Services): express.Application {
   app.use(setUpStaticResources())
   nunjucksSetup(app)
   app.use(setUpAuthentication())
+  app.use((req, res, next) => {
+    const hasJourneyId = validate(req.originalUrl.split('/')[1])
+    res.locals.auditEvent = {
+      pageNameSuffix: hasJourneyId
+        ? `${req.originalUrl.split('/')[1]}/${req.originalUrl.replace(/\?.*/, '').split('/').slice(2).join('/')}` // JOURNEYID_PAGE
+        : req.originalUrl.replace(/\?.*/, ''), // PAGE
+      who: res.locals.user.username,
+      correlationId: req.id,
+    }
+
+    res.prependOnceListener('close', async () => {
+      await services.auditService.logPageView(
+        req.originalUrl,
+        req.journeyData,
+        req.query,
+        res.locals.auditEvent,
+        `ACCESS_ATTEMPT_`,
+      )
+    })
+
+    const resRender = res.render
+    res.render = (view: string, options?) => {
+      type resRenderCb = (view: string, options?: object, callback?: (err: Error, html: string) => void) => void
+      ;(resRender as resRenderCb).call(res, view, options, async (err: Error, html: string) => {
```

Comment:
Not a fan of starting a line with `;` usually its considered a code smell in standard.
Would it work declaring something like `const resRender = res.render as (view: string, options?: object, callback?: (err: Error, html: string) => void) => void` ? Might have to use `as any` or `as unknown` first if not